index.js

```diff
@@ -1,254 +1,263 @@
         if (arguments.length === 1) return add(name, defaultComponent)
         return _set(name, component, options)
     }
 
-    function set(name, component, options) {
+    const set = (name, component, options) => {
         debug('Setting component %s on system %s', name, params.name)
         return _set(name, component, options)
     }
 
-    function remove(name) {
+    const remove = (name) => {
         debug('Removing component %s from system %s', name, params.name)
         delete definitions[name]
         return api
     }
 
-    function _set(name, component, options) {
+    const _set = (name, component, options) => {
         if (!component) throw new Error(format('Component %s is null or undefined', name))
-        definitions[name] = assign({}, options, { name: name, component: component.start ? component : wrap(component), dependencies: [] })
+        definitions[name] = Object.assign({}, options, {
+            name: name,
+            component: component.start ? component : wrap(component),
+            dependencies: []
+        })
         currentDefinition = definitions[name]
         return api
     }
 
-    function include(subSystem) {
+    const include = (subSystem) => {
         debug('Including definitions from sub system %s into system %s', subSystem.name, params.name)
-        definitions = assign({}, definitions, subSystem._definitions)
+        definitions = Object.assign({}, definitions, subSystem._definitions)
         return api
     }
 
+    //FIXME: there is some issue with implicit 'this' usage
     function dependsOn() {
         if (!currentDefinition) throw new Error('You must add a component before calling dependsOn')
-        currentDefinition.dependencies = toArray(arguments).reduce(toDependencyDefinitions, currentDefinition.dependencies)
+        currentDefinition.dependencies = [...arguments].reduce(toDependencyDefinitions, currentDefinition.dependencies)
         return api
     }
 
-    function toDependencyDefinitions(accumulator, arg) {
-        var record = typeof arg === 'string' ? { component: arg, destination: arg } : defaults({}, arg, { destination: arg.component })
+    const toDependencyDefinitions = (accumulator, arg) => {
+        const record = typeof arg === 'string' ? {
+            component: arg,
+            destination: arg
+        } : Object.assign({},{destination: arg.component}, arg);
         if (!record.component) throw new Error(format('Component %s has an invalid dependency %s', currentDefinition.name, JSON.stringify(arg)))
-        if (find(currentDefinition.dependencies, { destination: record.destination })) throw new Error(format('Component %s has a duplicate dependency %s', currentDefinition.name, record.destination))
+        if (currentDefinition.dependencies.find(dep => dep.destination === record.destination)) {
+            throw new Error(format('Component %s has a duplicate dependency %s', currentDefinition.name, record.destination))
```

Comment:
Any reason not to use use template strings if we're upgrading to ES6?

Reply:
Ironically, my initial purpose was to do a small PR and not include that change xD. But yes, if I do another PR for syntax I will include it for sure